src/components/Editor/CodeEditor.tsx

```diff
@@ -0,0 +1,211 @@
+import { AttachDocAction, loadDocAction } from '../../actions/docActions';
+import { ConnectionStatus, AddPeer, DisconnectPeer } from '../../actions/peerActions';
+
+import 'codemirror/lib/codemirror.css';
+import 'codemirror/theme/monokai.css';
+import './CodeEditor.css';
+
+type CodeEditorProps = {
+  docKey: string;
+};
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    width: '100%',
+    '& > * + *': {
+      marginTop: theme.spacing(2),
+    },
+  },
+}));
+
+export default function CodeEditor(props: CodeEditorProps) {
+  const dispatch = useDispatch();
+  const classes = useStyles();
+
+  const doc = useSelector((state: IAppState) => state.docState.doc);
+  const client = useSelector((state: IAppState) => state.docState.client);
+  const loading = useSelector((state: IAppState) => state.docState.loading);
+  const errorMessage = useSelector((state: IAppState) => state.docState.errorMessage);
+  const peerClients = useSelector((state: IAppState) => state.peerState.peers);
+  const otherClientsCursor = useRef<Map<string, ClientCursor>>(new Map());
+
+  const connectClient = (clientId: string) => {
+    const existedClient = peerClients[clientId];
+
+    let color: string;
+    if (existedClient && existedClient.status === ConnectionStatus.Disconnected) {
+      color = existedClient.color;
+    } else {
+      color = randomColor();
+    }
+
+    const newClientCursor = ClientCursor.of(clientId, color);
+    otherClientsCursor.current.set(clientId, newClientCursor);
+    dispatch(AddPeer(clientId, color));
+  };
+
+  // Attach document
+  useEffect(() => {
+    dispatch(loadDocAction(true));
+    dispatch(AttachDocAction(props.docKey));
+  }, [props.docKey, dispatch]);
+
+  // Subscribe other client
+  useEffect(() => {
+    if (!client || !doc) {
+      return;
+    }
+
+    const disconnectClient = (clientId: string) => {
+      if (otherClientsCursor.current.has(clientId)) {
+        otherClientsCursor.current.get(clientId)!.removeCursor();
+        otherClientsCursor.current.delete(clientId);
+      }
+      dispatch(DisconnectPeer(clientId));
+    };
+
+    const unsubscribe = client.subscribe((event: any) => {
+      if (event.name === 'documents-watching-peer-changed') {
+        const newPeerClientsId: string[] = event.value[doc.getKey().toIDString()];
+        const setNewPeerClientsId = new Set(newPeerClientsId);
+
+        for (const clientId of Object.keys(peerClients)) {
+          if (setNewPeerClientsId.has(clientId) && peerClients[clientId].status === ConnectionStatus.Connected) {
+            continue;
+          }
+          disconnectClient(clientId);
+        }
+      }
+    });
+
+    return () => {
+      unsubscribe();
+    };
+  }, [client, doc, peerClients, dispatch]);
+
+  if (loading) {
+    return (
+      <Box height="100%">
+        <CircularProgress color="secondary" />
+      </Box>
+    );
+  }
+
+  if (errorMessage || client === null || doc === null) {
```

Comment:
```suggestion
  if (errorMessage || !client || !doc) {
```
How about making this modification? Tell us what you think

Reply:
I saw this suggestion in [Airbnb JavaScript Style Guide](https://github.com/airbnb/javascript#comparison--ifl). We don't yet have a style guide for TypeScript, so I think we have to decide on one thing.

Do you have any suggestion style guide?

Reply:
@ppeeou @dc7303 
https://basarat.gitbook.io/typescript/styleguide#null-vs-undefined
Here's a better guide than what I suggested.
And I think this guide looks pretty good, so I'll review it several times.
Please read it and give us your opinion :)

Reply:
@dc7303 Thanks for the great suggestion.

@dc7303 @ppeeou 
Of the reviews in this PR, there were more style reviews than logic. The reason seems to be that `Linter` was not introduced. First of all, I think it is good to skip the style review of this PR and create a separate issue introducing Linter.